src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_VALUE_FOR_RANDOM = 20;
+    private static final double DEFAULT_RADIUS_CIRCLE = 10.0;
+    private static final int NUMBERS_OF_FIGURES = 4;
+    private Random random = new Random();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomFigure() {
```

Comment:
Do not create 4 figures. You do extra work - you need only one figure.
1. Make your decision
2. Create figure of desired type
Also, better add private methods (like `getRandomSquare()`) for creation.